src/wavm/wavm.cc

```diff
@@ -272,21 +293,44 @@ bool Wavm::load(const std::string &code, bool allow_precompiled) {
   return true;
 }
 
-AbiVersion Wavm::getAbiVersion() { return AbiVersion::Unknown; }
+AbiVersion Wavm::getAbiVersion() {
+  if (abi_version_ != AbiVersion::Unknown) {
+    return abi_version_;
+  }
+  for (auto &e : ir_module_.exports) {
+    if (e.name == "proxy_abi_version_0_1_0") {
+      abi_version_ = AbiVersion::ProxyWasm_0_1_0;
+      return abi_version_;
+    }
+    if (e.name == "proxy_abi_version_0_2_0") {
+      abi_version_ = AbiVersion::ProxyWasm_0_2_0;
+      return abi_version_;
+    }
+  }
```

Comment:
Please add the check for `proxy_abi_version_0_2_1` as well.

(We should probably add a utility function `(string_view function_name) -> AbiVersion` instead of replicating this check in both runtimes, but that can be separate PR or issue).

Reply:
Done.  Agreed.